workitem/workitem_repository.go

```diff
@@ -544,24 +544,11 @@ func (r *GormWorkItemRepository) Reorder(ctx context.Context, spaceID uuid.UUID,
 	default:
 		return &wi, nil
 	}
-	res.Version = res.Version + 1
-	res.Type = wi.Type
-	res.Fields = Fields{}
 
-	res.ExecutionOrder = order
-
-	for fieldName, fieldDef := range wiType.Fields {
-		if fieldDef.ReadOnly {
-			continue
-		}
-		fieldValue := wi.Fields[fieldName]
-		var err error
-		res.Fields[fieldName], err = fieldDef.ConvertToModel(fieldName, fieldValue)
-		if err != nil {
-			return nil, errors.NewBadParameterError(fieldName, fieldValue)
-		}
-	}
-	tx = tx.Where("Version = ?", wi.Version).Save(&res)
+	tx = tx.Model(&res).Select("execution_order", "version").Updates(WorkItemStorage{
```

Comment:
Why doesn't this have a `where` clause?

Reply:
because I pass `.Model` instead of `.Where`

Reply:
@DhritiShikhar can you please show the `.Debug().` output of this query? I think Gorm only checks for the primary key if you do `.Model(&res).`. And the `version` is not the primary field.